control-plane-operator/controllers/hostedcontrolplane/multus_disable_test.go

```diff
@@ -0,0 +1,165 @@
+			} else {
+				assert.NotNil(t, hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork, test.description)
+				assert.Equal(t, *test.expectedDisableMultiNetwork, *hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork, test.description)
+			}
+
+			// Test that the reconciler logic would make the correct decision
+			// When DisableMultiNetwork is nil, it defaults to false (multus enabled)
+			disableMultiNetwork := false
+			if hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork != nil {
+				disableMultiNetwork = *hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork
+			}
+			shouldProcessMultus := !disableMultiNetwork
+
+			expectedDisableMultiNetwork := false
+			if test.expectedDisableMultiNetwork != nil {
+				expectedDisableMultiNetwork = *test.expectedDisableMultiNetwork
+			}
+			expectedShouldProcess := !expectedDisableMultiNetwork
+			assert.Equal(t, expectedShouldProcess, shouldProcessMultus,
+				"Reconciler logic should correctly interpret DisableMultiNetwork field")
+		})
+	}
+}
+
+func TestDisableMultiNetworkRestartLogic(t *testing.T) {
+	tests := []struct {
+		name                 string
+		disableMultiNetwork  *bool
+		hasRestartAnnotation bool
+		expectedRestart      bool
+	}{
+		{
+			name:                 "Should restart: has annotation and multus enabled",
+			disableMultiNetwork:  ptr.To(false),
+			hasRestartAnnotation: true,
+			expectedRestart:      true,
+		},
+		{
+			name:                 "Should not restart: has annotation but multus disabled",
+			disableMultiNetwork:  ptr.To(true),
+			hasRestartAnnotation: true,
+			expectedRestart:      false,
+		},
+		{
+			name:                 "Should not restart: no annotation, multus enabled",
+			disableMultiNetwork:  ptr.To(false),
+			hasRestartAnnotation: false,
+			expectedRestart:      false,
+		},
+		{
+			name:                 "Should not restart: no annotation, multus disabled",
+			disableMultiNetwork:  ptr.To(true),
+			hasRestartAnnotation: false,
+			expectedRestart:      false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			hcp := &hyperv1.HostedControlPlane{
+				Spec: hyperv1.HostedControlPlaneSpec{
+					OperatorConfiguration: &hyperv1.OperatorConfiguration{ClusterNetworkOperator: &hyperv1.ClusterNetworkOperatorSpec{DisableMultiNetwork: test.disableMultiNetwork}},
+				},
+			}
+
+			// Test the restart logic: restart when hasRestartAnnotation && !DisableMultiNetwork
+			// When DisableMultiNetwork is nil, it defaults to false (multus enabled)
+			disableMultiNetwork := false
+			if hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork != nil {
+				disableMultiNetwork = *hcp.Spec.OperatorConfiguration.ClusterNetworkOperator.DisableMultiNetwork
+			}
+			shouldRestart := test.hasRestartAnnotation && !disableMultiNetwork
```

Comment:
again, for this to be a meaningful test, it should be calling code from the hostedcontrolplane_controller.go
Maybe extract that logic into function called setMultusAdmissionControllerRestartAnnotation and call that function from this test?